### ISSUE TITLE
False N+1 reported in tests after upgrade to Rails 5.2

### ISSUE THREAD

**Issue by edzhelyov** [flyerhzm/bullet#416]
Bullet version 5.7.5. Everything is fine on the current Rails 5.1.6 version. But when I upgrade Rails to 5.2 I have a test that starts failing with:
```
Failure/Error: Bullet.perform_out_of_channel_notifications if Bullet.notification?

     Bullet::Notification::UnoptimizedQueryError:
       user: xxx

       USE eager loading detected
         Receipt => [:account]
         Add to your finder: :includes => [:account]
       Call stack
```

The test looks something like:
```
...
  let(:account) { create :account }
  it 'works' do
    params = { account_id: account.id, name: 'Receipt' }
    CreateReceipt.execute params
  end
```
And receipt in the fashion of:
```
class Receipt < ApplicationRecord
  after_create :create_callback

  def create_callback
    Event.created self
  end
end

class Event
  def self.created(receipt)
    receipt.account.name
    etc...
  end
end
```

In Rails 5.2 the reference to the `receipt.account` in `Event::created` starts to throw N+1. Now `receipt.account` is not loaded until then, but there is no multiple record loaded in the tests, so it's some kind of an issue. I'm trying to debug that, but my Bullet knowledge is 0, so any help will be appreciated.

**Comment by cabello** [comment]
I am on the same boat, looking at my traces it does not look like the 2, 3 queries I am seeing should be considered N+1 queries.

**Comment by flyerhzm** [comment]
@edzhelyov could you please give a try to bullet latest master branch?

**Comment by edzhelyov** [comment]
Latest master solves my problems. Seems this is the commit that fixes it: https://github.com/flyerhzm/bullet/commit/49352b0f948837d690f5d9bfdaa9e89c57d911a8.

Can you release a minor version? It's quite a annoying issue and I think it requires a version on it's own.

**Comment by cabello** [comment]
Hm... why did you close if there is no new release yet?

**Comment by cabello** [comment]
@flyerhzm 👋 how long do you think it will take for Rubygems to fetch the 5.7.6 release?

**Comment by flyerhzm** [comment]
@edzhelyov @cabello I just released version `5.7.6`.

**Comment by johan-smits** [comment]
To me the issues was not Rails 5.2 related because I was already on the 5.2.0 for some time.
Even with the 5.7.6 release I get the false positive. I run now Rails 5.2.1 and 5.7.5 for a working combination.

Current master af60d11a8904aef6477e257aa7ca165927c28469 is also broken for me.
